### PR TITLE
discovery improvement work; sync solr configs with sul-solr-config

### PR DESCRIPTION
**Pull request by ndushay**
~~HOLD for integration tests to pass on stage.~~

# Why was this change made?

Part of sul-dlss/dor_indexing_app#1032

As part of the Argo Discovery Improvements, various needs have arisen:
- improve search results for queries on authors, source ids, and in general (catch all field) with qf changes in blacklight_config_helper
- sync Solr config files with updates in  sul-dlss/sul-solr-configs/pull/280

Andrew vetted the blacklight_config_helper changes when I set him up with the same configuration for `qt=qttest` in the url;  he vetted the improvements for author searching and for source id searching.

I diffed solrconfig.xml against the ones in sul-solr-configs/pull/281 and they are the same in all the right ways

# How was this change tested?
- ci tests
- see sul-dlss/sul-solr-configs/pull/280 for how the Solr config changes were tested

TODO:
- [x] deploy branch to -stage and run integration tests
- [x] get this PR merged
- [ ] deploy to all environments (intent is for this to happen with dependency updates on Monday)

### CODE REVIEW

**Review by justinlittman — APPROVED** [review]

File: app/controllers/catalog_controller.rb [hunk 1]
```diff
@@ -240,11 +240,11 @@ def index
           contributor_text_nostem_im^3
           topic_tesim^2
 
-          exploded_project_tag_ssim^2
-          exploded_nonproject_tag_ssim
-          exploded_tag_ssim
```
Comment (ndushay) [inline app/controllers/catalog_controller.rb:245]: qttest fields searched:  tag_ssim is sufficient for searching.  "exploded" tag fields are for hierarchical facets

File: app/helpers/blacklight_config_helper.rb [hunk 2]
```diff
@@ -6,18 +6,38 @@ def self.add_common_default_solr_params_to_config!(config)
...
+        identifier_tesim
+        barcode_id_ssim
+        folio_instance_hrid_ssim
         source_id_ssim
-        tag_ssim
-        topic_tesim
+        source_id_ssi
+        source_id_text_nostem_i^3
+        previous_ils_ids_ssim
+        doi_ssim
```
Comment (ndushay) [inline app/helpers/blacklight_config_helper.rb:39]: this changes which Solr fields Argo searches.  We search all the old fields, and have added some to be "more like SearchWorks":
- author names, unstemmed (names should rarely be stemmed!)
- catch-all fields descriptive_t** (any reasonable descriptive metadata values)
- format, content type, object type
- all ids, all flavors, all the time

File: app/controllers/catalog_controller.rb [hunk 3]
```diff
@@ -240,11 +240,11 @@ def index
...
           tag_ssim
 
+          originInfo_place_placeTerm_tesim
+          originInfo_publisher_tesim
```
Comment (ndushay) [inline app/controllers/catalog_controller.rb:246]: qttest fields searched:  publisher and place should have been here all along

File: solr_conf/conf/schema.xml [hunk 4]
```diff
@@ -8,37 +8,61 @@
 
   <fields>
     <field name="id" type="string" stored="true" indexed="true" multiValued="false" required="true"/>
+    <!-- DEPRECATED: "long" is a Trie based field type; use solr.LongPointField class instead ... -->
```
Comment (ndushay) [inline solr_conf/conf/schema.xml:11]: See https://github.com/sul-dlss/sul-solr-configs/pull/280/files for rationale behind schema.xml changes

File: solr_conf/conf/solrconfig.xml [hunk 5]
```diff
@@ -35,16 +35,86 @@
       <str name="fl"> *, score</str>
 
       <str name="qf">
-        id
-        descriptive_tiv
+        sw_display_title_tesim^5
```
Comment (ndushay) [inline solr_conf/conf/solrconfig.xml:38]: make the default qf here match what is in blacklight_config_helper qf (the latter of these is what is actually used for qf)

File: solr_conf/conf/solrconfig.xml [hunk 6]
```diff
@@ -35,16 +35,86 @@
...
+        source_id_ssim^3
+        source_id_ssi
+        source_id_text_nostem_i^3
+        previous_ils_ids_ssim
+        doi_ssim
+        contributor_orcids_ssim
       </str>
-      <str name="pf">
-        descriptive_tiv^10
+      <str name="pf">  <!-- (phrase boost within result set) -->
```
Comment (ndushay) [inline solr_conf/conf/solrconfig.xml:71]: pf, pf3 and pf2 are about boosting phrase results within the result set ... and we want to do this!  Not sure why it was left off before.

File: solr_conf/conf/solrconfig.xml [hunk 7]
```diff
@@ -76,26 +146,27 @@
...
       <str name="qf">
         sw_display_title_tesim^5
-        sw_author_tesim^3
+        contributor_text_nostem_im^3
         topic_tesim^2
 
-        exploded_project_tag_ssim^2
-        exploded_nonproject_tag_ssim
-        exploded_tag_ssim
         tag_ssim
```
Comment (ndushay) [inline solr_conf/conf/solrconfig.xml:152]: tag_ssim is sufficient for searching. "exploded" tag fields are for hierarchical facets

File: solr_conf/conf/solrconfig.xml [hunk 8]
```diff
@@ -76,26 +146,27 @@
...
 
+        originInfo_place_placeTerm_tesim
+        originInfo_publisher_tesim
```
Comment (ndushay) [inline solr_conf/conf/solrconfig.xml:155]: leaving publisher and place out before was an oversight

File: solr_conf/conf/solrconfig.xml (hunk 5 above)
Comment (ndushay) [reply solr_conf/conf/solrconfig.xml:38]: See PR sul-dlss/sul-solr-configs/pull/281 for more info on these changes

**Review by aaron-collier — APPROVED** [review]

**Pull request merged**